.circleci/config.yml

```diff
@@ -1,44 +1,297 @@
+      - singularity-tests:
+          <<: *common_filters
+          requires:
+            - build
+
+
+jobs:
+  unit-tests:
+    executor: tests_executor
+    environment:
+      TESTING_PROFILE: unit-tests
+    steps:
+      - setup_for_tests
+      - run_tests
+      - clean_and_save_cache
+      - save_test_results
+      - send_coverage
+
+  singularity-tests:
+    # Must use machine instead of Docker for Singularity
+    executor: machine_integration_test_exec
+    environment:
+      TESTING_PROFILE: singularity-tests
+    steps:
+      - setup_machine
+      - setup_for_tests
+      - setup_machine_env_vars
+      - setup_for_integration_tests
+      - run_tests
+      - clean_and_save_cache
+      - save_test_results
+      - send_coverage
+
+  non-confidential-tests:
+    # We must use machine executor instead of Docker executor and setup_remote_docker
+    # https://circleci.com/docs/2.0/executor-intro/#machine
+    # perhaps because cwltool cannot mount a volume from our job space
+    # into CircleCI remote Docker
+    # https://circleci.com/docs/2.0/building-docker-images/#mounting-folders
+    executor: machine_integration_test_exec
+    environment:
+      TESTING_PROFILE: non-confidential-tests
+    steps:
+      - setup_machine
+      - setup_for_tests
+      - setup_machine_env_vars
+      - setup_for_integration_tests
+      - run_tests
+      - clean_and_save_cache
+      - save_test_results
+      - send_coverage
+
+  confidential-workflow-integration-tests:
+    # We must use machine executor instead of Docker executor and setup_remote_docker
+    # https://circleci.com/docs/2.0/executor-intro/#machine
+    # perhaps because cwltool cannot mount a volume from our job space
+    # into CircleCI remote Docker
+    # https://circleci.com/docs/2.0/building-docker-images/#mounting-folders
+    executor: machine_integration_test_exec
+    environment:
+      TESTING_PROFILE: confidential-workflow-tests
+    steps:
+      - setup_machine
+      - setup_for_tests
+      - setup_machine_env_vars
+      - setup_for_integration_tests
+      - run_tests
+      - clean_and_save_cache
+      - save_test_results
+      - send_coverage
+
+  confidential-tool-integration-tests:
+    # We must use machine executor instead of Docker executor and setup_remote_docker
+    # https://circleci.com/docs/2.0/executor-intro/#machine
+    # perhaps because cwltool cannot mount a volume from our job space
+    # into CircleCI remote Docker
+    # https://circleci.com/docs/2.0/building-docker-images/#mounting-folders
+    executor: machine_integration_test_exec
+    environment:
+      TESTING_PROFILE: confidential-tool-tests
+    steps:
+      - setup_machine
+      - setup_for_tests
+      - setup_machine_env_vars
+      - setup_for_integration_tests
+      - run_tests
+      - clean_and_save_cache
+      - save_test_results
+      - send_coverage
+
+
+    # This catagory seems to be covered by confidential and non-confidential tests
```

Comment:
Can delete and seems reasonable that it would be covered by confidential vs non-confidential

Reply:
Deleted commented out code